OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -80,6 +80,10 @@ protected virtual bool CanAttack(Actor self, Target target)
 			if (!HasAnyValidWeapons(target))
 				return false;
 
+			var layer = self.Location.Layer;
+			if (layer == CustomMovementLayerType.Tunnel || layer == CustomMovementLayerType.Subterranean)
```

Comment:
There's one other thing I could try to get the desired behavior, namely treating targets on non-interactable layers as if they were 'out of range' in the `Attack` activity. That would be a little hacky, though.